src/main/java/com/google/step/servlets/DealDetailServlet.java

```diff
@@ -46,13 +59,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
       return;
     }
-    Deal deal = manager.readDeal(id);
+    Deal deal = dealManager.readDeal(id);
     if (deal == null) {
       response.setStatus(HttpServletResponse.SC_NOT_FOUND);
       return;
     }
+
+    // TODO get real restaurant
+    Restaurant restaurant = new Restaurant(deal.restaurantId, "Restaurant Name", "ablobkey");
+
+    User poster = userManager.readUser(deal.posterId);
+
+    // TODO get real tags
+    List<Tag> tags = new ArrayList<>();
+
+    int votes = voteManager.getVotes(deal.id);
+
     response.setContentType("application/json;");
-    response.getWriter().println(JsonFormatter.getDealJson(deal));
+    response.getWriter().println(JsonFormatter.getDealJson(deal, restaurant, poster, tags, votes));
```

Comment:
Should there be a test for these changes?

Reply:
I added a test for JsonFormatter. The servlet test can be done in another PR.